libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -402,15 +403,24 @@ public override async Task<ResourceResponse[]> SendActivitiesAsync(ITurnContext
                 {
                     // if it is a Trace activity we only send to the channel if it's the emulator.
                 }
-                else if (!string.IsNullOrWhiteSpace(activity.ReplyToId))
-                {
-                    var connectorClient = turnContext.TurnState.Get<IConnectorClient>();
-                    response = await connectorClient.Conversations.ReplyToActivityAsync(activity, cancellationToken).ConfigureAwait(false);
-                }
                 else
                 {
-                    var connectorClient = turnContext.TurnState.Get<IConnectorClient>();
-                    response = await connectorClient.Conversations.SendToConversationAsync(activity, cancellationToken).ConfigureAwait(false);
+                    if (!activity.ServiceUrl.StartsWith("http"))
```

Comment:
I'm skeptical of this code. It seems really fragile. 

Can you add in a more detailed explication of what's going on?

Reply:
This should be centralized - you already ahve the same code in the RequiresLink method. Move this logic to a small class that takes an activity and determines the type of connection, and has meaningful documentation and names that allow the code to be more easily understandable by future maintainers. Thanks!

Reply:
This is all well and good, and I'll do that...but because Prompts and Adapters live in different assemblies I will now have to make this method public, which means when the rules change in the future this method will change.